src/client/tile.cpp

```diff
@@ -1735,6 +1735,43 @@ bool TextureSource::generateImagePart(std::string part_of_name,
 			baseimg->drop();
 			baseimg = image;
 		}
+		/*
+			[opacity:R
+			Makes the base image transparent according to the given ratio.
+			R must be between 0 and 255.
+			0 means totally transparent.
+			255 means totally opaque.
+		*/
+		else if (str_starts_with(part_of_name, "[opacity:"))
+		{
+			if (baseimg == NULL) {
+				errorstream << "generateImagePart(): baseimg == NULL "
+						<< "for part_of_name=\"" << part_of_name
+						<< "\", cancelling." << std::endl;
+				return false;
+			}
+
+			Strfnd sf(part_of_name);
+			sf.next(":");
+			std::string ratio_str = sf.next("");
+
+			int ratio = 0;
+
+			if (is_number(ratio_str))
+				ratio = mystoi(ratio_str, 0, 255);
+
+			core::dimension2d<u32> dim = baseimg->getDimension();
+
+			for (u32 y = 0; y < dim.Height; y++)
+			for (u32 x = 0; x < dim.Width; x++)
+			{
+				video::SColor c = baseimg->getPixel(x,y);
+				if (c.getAlpha() != 0) {
+					c.setAlpha(ratio);
```

Comment:
I tried an algorithm that does not make much sense, it "softly" increases and decreases the alpha values in between:
```
Alpha mod:      Image alpha:    Result:
64              255             64.00
64              128             0.00
64              64              0.00
64              0               0.00
128             255             128.00
128             128             15.97
128             64              0.00
128             0               0.00
255             255             255.00
255             128             221.12
255             64              168.95
255             0               0.00

Formula in Lua: math.sqrt(math.max(0, ALPHA_MOD^2 - (255-BASE_ALPHA)^2))
where ALPHA_MOD = ratio and BASE_ALPHA = current pixel alpha
```
Not sure if this is too much calculating for a simple opacity modifier.